pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/buffer/TopicTransactionBufferTest.java

```diff
@@ -179,4 +182,57 @@ public void testCloseTransactionBufferWhenTimeout() throws Exception {
         Assert.assertTrue(f.isCompletedExceptionally());
     }
 
+    /**
+     * This test verifies the state changes of a TransactionBuffer within a topic under different conditions.
+     * Initially, the TransactionBuffer is in a NoSnapshot state upon topic creation.
+     * It remains in the NoSnapshot state even after a normal message is sent.
+     * The state changes to Ready only after a transactional message is sent.
+     * The test also ensures that the TransactionBuffer can be correctly recovered after the topic is unloaded.
+     */
+    @Test
+    public void testWriteSnapshotWhenFirstTxnMessageSend() throws Exception {
+        String topic = "persistent://" + NAMESPACE1 + "/testWriteSnapshotWhenFirstTxnMessageSend";
+        admin.topics().createNonPartitionedTopic(topic);
+        PersistentTopic persistentTopic = (PersistentTopic) pulsarServiceList.get(0).getBrokerService()
+                .getTopic(topic, false)
+                .get()
+                .get();
+        persistentTopic.checkIfTransactionBufferRecoverCompletely(true).get();
+        TopicTransactionBuffer topicTransactionBuffer = (TopicTransactionBuffer) persistentTopic.getTransactionBuffer();
+        // The TransactionBuffer should be in NoSnapshot state when the topic is initially created
+        Assert.assertEquals(topicTransactionBuffer.getState(), TopicTransactionBufferState.State.NoSnapshot);
```

Comment:
Maybe use Awaitility will better?

```java
TopicTransactionBuffer topicTransactionBuffer = (TopicTransactionBuffer) persistentTopic.getTransactionBuffer();
Awaitility.await().untilAsserted(() -> {
    Assert.assertEquals(topicTransactionBuffer.getState(), TopicTransactionBufferState.State.NoSnapshot);
});
```

Reply:
There is a wait `persistentTopic.checkIfTransactionBufferRecoverCompletely(true).get();` before this check.

Reply:
I checked the real logic, the `persistentTopic.checkIfTransactionBufferRecoverCompletely(true).get();` is only used in the test, so I think using Awaitility until the state is `NoSnapshot ` will be better. WDYT?

Reply:
Fixed.